### DISCUSSION TITLE
Sampling using Javaagent

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#5803] [Q&A]
Hi,
Can you please let me know the steps to enable Sampling using Javaagent.
We were using Jaeger sampling (JAEGER_SAMPLER_TYPE, JAEGER_SAMPLING_PARAM variables) using OpenTracing earlier and now migrating to OpenTelemetry.

**Comment by trask** [comment]
hi @itsmaneesh, check out https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler

**Reply by itsmaneesh** [reply]
@trask I understand that OpenTelemetry expects the below properties for Sampling to take place:
OTEL_TRACES_SAMPLER
OTEL_TRACES_SAMPLER_ARG
Our requirement is to have Sampling strategy as [type: ratelimiting, param: 2.0] i.e. Sample request with the rate of 2 traces per second.
I couldn't find clear mapping for [Sampler Types ](https://www.jaegertracing.io/docs/1.33/sampling/#client-sampling-configuration)
 like `const`, `probabilistic`, `ratelimiting` and `remote` details in the OpenTelemetry auto configure [README](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler))

So, Can you please clarify whether the below configuration works for our case: 
OTEL_TRACES_SAMPLER = parentbased_traceidratio
OTEL_TRACES_SAMPLER_ARG = 2.0

Thank you.

**Reply by trask** [reply]
hi @itsmaneesh, if you are looking for a ratelimiting sampler, check out ConsistentRateLimitingSampler at https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/consistent-sampling

**Reply by itsmaneesh** [reply]
@trask Thank you for the input. 
I have couple of follow up questions on this consistent-sampling:
1. ConsistentRateLimitingSampler seems to be sampling the spans per second as per `targetSpansPerSecondLimit` method param value. 
    
      for e.g.: In our use case number of spans per request will vary. So, if we define `targetSpansPerSecondLimit` value as 5 then we 
      would miss the spans if there are more than 5 spans per second and that would not give us the complete information.
      So, we are looking for a Rate limiting Sampling strategy that samples the `traces` per second so that we sample the number of 
      traces and will have the complete spans information.

2. I couldn't come across a documentation that explains how to use this `consistent-sampling` project. Could you please guide me to a documentation that provide more insights to how to use this project.

**Reply by itsmaneesh** [reply]
@trask  Can you please provide your clarifications to the above questions.
Thanks.

**Reply by itsmaneesh** [reply]
@trask Can you please provide clarifications to the above questions.
